app/assets/javascripts/preferences/panes/two-factor-auth/TwoFactorActivation.ts

```diff
@@ -0,0 +1,125 @@
+import { action, makeAutoObservable, observable, untracked } from 'mobx';
+import { MfaGateway } from './MfaProps';
+
+const activationSteps = [
+  'scan-qr-code',
+  'save-secret-key',
+  'verification',
+] as const;
+
+type ActivationStep = typeof activationSteps[number];
+
+export class TwoFactorActivation {
+  public readonly type = 'two-factor-activation' as const;
+
+  private _activationStep: ActivationStep;
+
+  private _2FAVerification: 'none' | 'invalid' | 'valid' = 'none';
+
+  private inputSecretKey = '';
+  private inputOtpToken = '';
+
+  constructor(
+    private mfa: MfaGateway,
+    private readonly _secretKey: string,
+    private _cancelActivation: () => void,
+    private _enabled2FA: () => void
+  ) {
+    this._activationStep = 'scan-qr-code';
+
+    makeAutoObservable<
+      TwoFactorActivation,
+      | '_secretKey'
+      | '_authCode'
+      | '_step'
+      | '_enable2FAVerification'
+      | 'refreshOtp'
+      | 'inputOtpToken'
+      | 'inputSecretKey'
+    >(
+      this,
+      {
+        _secretKey: observable,
+        _authCode: observable,
+        _step: observable,
+        _enable2FAVerification: observable,
+        refreshOtp: action,
+        inputOtpToken: observable,
+        inputSecretKey: observable,
+      },
+      { autoBind: true }
+    );
+  }
+
+  get secretKey() {
+    return this._secretKey;
+  }
+
+  get activationStep() {
+    return this._activationStep;
+  }
+
+  get verificationStatus() {
+    return this._2FAVerification;
+  }
+
+  get qrCode() {
+    return `otpauth://totp/2FA?secret=${this._secretKey}&issuer=Standard%20Notes`;
```

Comment:
(we'd need to get user email into this function)

```suggestion
    return `otpauth://totp/2FA?secret=${this._secretKey}&issuer=Standard%20Notes&label=${user.email}`;
```

This wasn't present in MFA Link because it didn't have access to user email, but we do here.

Reply:
Done.